docs/tools/testing.ipynb

```diff
@@ -0,0 +1,57 @@
+{
+ "cells": [
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "title: Testing\n",
+    "sidebar_position: 2\n",
+    "description: Testing Mojo programs.\n",
+    "css: /static/styles/page-navigation.css\n",
+    "website:\n",
+    "  open-graph:\n",
+    "    image: /static/images/mojo-social-card.png\n",
+    "  twitter-card:\n",
+    "    image: /static/images/mojo-social-card.png\n",
+    "---"
+   ],
+   "id": "e622ec78c00f1839"
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Built-in test runner\n",
+    "\n",
+    "## File and folder conventions\n",
+    "\n",
+    "The Mojo CLI has a built-in test runner that can be used to test Mojo programs. `mojo test` will recursively search the `test` directory for any files with `test` in the filename. So in the following example:\n",
```

Comment:
FYI @River707.  We intentionally haven't documented `mojo test` yet since there are some things that need to get wrapped up still before it's officially ready for public use.

Reply:
I didn't know. Sorry.

Reply:
No worries at all @KarateCowboy! I'll work with @arthurevans and @River707 and we'll figure this out.  Don't feel bad about documenting something that's out there and available for us.  I didn't mean to imply that; this is just something we were knowingly sitting on a bit (`mojo test`) while we polish things and then do eventual more documentation, blog post, etc on it all.

Reply:
Cool beans, cool beans. Don't feel bad just tossing the commit. My only ask is you take it to heart as a sign of how excited you have some devs out there -- one of them sat down and mapped it all out for all the other TDD fanatics a-chomping at the bit for this beautiful child of Python and Rust. I honestly didn't know that I was making a Jupyter notebook -- only that I needed to make a file like that to update the docs!

Reply:
We really appreciate your dedication and excitement! ❤️ ❤️ ❤️ 

I'm a huge fan of TDD personally and `mojo test` will be a big improvement 😄 .